pynest/nest/tests/test_stdp_synapse.py

```diff
@@ -0,0 +1,360 @@
+# -*- coding: utf-8 -*-
+#
+# test_stdp_synapse.py
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+import nest
+import unittest
+from math import exp
+import numpy as np
+
+try:
+    import matplotlib as mpl
+    import matplotlib.pyplot as plt
+    DEBUG_PLOTS = True
+except Exception:
+    DEBUG_PLOTS = False
```

Comment:
I am not sure that producing plots just because matplotlib is available is a good idea. We do not need plots when running CI, only when debugging. Wouldn't it be better to either make this switch manual (by editing the file)?

See also the "plotting possible" test in `test_visualisation`.

Reply:
The "plotting possible" test in ``test_visualization`` does essentially the same thing. matplotlib is not installed on CI, as far as I understood, that is by design.